Alignment/APEEstimation/plugins/ApeTreeCreateDefault.cc

```diff
@@ -0,0 +1,473 @@
+    if(1==*i_id && id)return true;
+    if(2==*i_id && !id)return true;
+  }
+  return false;
+}
+
+bool
+ApeTreeCreateDefault::checkModuleDirections(const int id, const std::vector<int>& v_id)const
+{
+    
+  if(v_id.empty())return true;
+  for(std::vector<int>::const_iterator i_id = v_id.begin(); i_id != v_id.end(); ++i_id){
+    if(id==*i_id)return true;
+  }
+  return false;
+}
+
+bool
+ApeTreeCreateDefault::checkModulePositions(const float id, const std::vector<double>& v_id)const
+{
+    
+  if(v_id.empty())return true;
+  int entry(1); double intervalBegin(999.);
+  for(std::vector<double>::const_iterator i_id = v_id.begin(); i_id != v_id.end(); ++i_id, ++entry){
+    if(entry%2==1)intervalBegin = *i_id;
+    if(entry%2==0 && id>=intervalBegin && id<*i_id)return true;
+  }
+  return false;
+}
+
+
+
+// ------------ method called to for each event  ------------
+void
+ApeTreeCreateDefault::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  // Same procedure as in ApeEstimatorSummary.cc minus reading of baseline tree
+  
+  // Load APEs from the GT and write them to root files similar to the ones from calculateAPE()
+  edm::ESHandle<AlignmentErrorsExtended> alignmentErrors;
+  iSetup.get<TrackerAlignmentErrorExtendedRcd>().get(alignmentErrors);  
+  
+  // Set up root file for default APE values
+  const std::string defaultFileName(parameterSet_.getParameter<std::string>("resultFile"));
+  TFile* defaultFile = new TFile(defaultFileName.c_str(),"RECREATE");
+  
+  // Naming in the root files has to be iterTreeX to be consistent for the plotting tool
+  TTree* defaultTreeX(nullptr);
+  TTree* defaultTreeY(nullptr);
+  defaultFile->GetObject("iterTreeX;1",defaultTreeX);
+  defaultFile->GetObject("iterTreeY;1",defaultTreeY);
+  // The same for TTree containing the names of the sectors (no additional check, since always handled exactly as defaultTree)
+  TTree* sectorNameTree(nullptr);
+  defaultFile->GetObject("nameTree;1",sectorNameTree);
+  
+  edm::LogInfo("DefaultAPETree")<<"APE Tree is being created";
+  defaultTreeX = new TTree("iterTreeX","Tree for default APE x values from GT");
+  defaultTreeY = new TTree("iterTreeY","Tree for default APE y values from GT");     
+  sectorNameTree = new TTree("nameTree","Tree with names of sectors");
+  
+    
+  // Assign the information stored in the trees to arrays
+  std::vector<double*> a_defaultSectorX;
+  std::vector<double*> a_defaultSectorY;
+  
+  std::vector<std::string*> a_sectorName;
+  std::map<unsigned int, TrackerSectorStruct>::const_iterator i_sector;
+  for(i_sector = m_tkSector_.begin(); i_sector != m_tkSector_.end(); ++i_sector){
+    const unsigned int iSector(i_sector->first);
+    const bool pixelSector(i_sector->second.isPixel);
+    
+    a_defaultSectorX.push_back(new double(-99.));
+    a_defaultSectorY.push_back(new double(-99.));
+    a_sectorName.push_back(new std::string(i_sector->second.name));
```

Comment:
I see a lot of `new` statements but no matching `delete`. For the objects in the TFile, ROOT will take care of when closing the file (you should also delete the file), but not for the objects you store in the vector.
I know that for this tool, memory is not really an issue, but for a new tool, we should spend some time to carefully code it.